include/sparrow/buffer_view.hpp

```diff
@@ -116,6 +120,17 @@ namespace sparrow
     {
     }
 
+    template <class T>
+    template <class It>
+        requires std::input_iterator<It>
+                     && std::same_as<std::remove_const_t<std::iter_value_t<It>>, std::remove_const_t<T>>
+    buffer_view<T>::buffer_view(It first, It last)
+        : p_data(&*first)
```

Comment:
I dont think that's working if `It` is not a contiguous iterator?

BTW there is `addressof` and `pointer_to` that cold help with that expression but I'm not totally sure if it's necessary here.

Reply:
Adressed too